src/Forms/HTMLEditor/HTMLEditorField.php

```diff
@@ -185,4 +185,16 @@ public function Field($properties = array())
         $this->getEditorConfig()->init();
         return parent::Field($properties);
     }
+
+    public function getSchemaStateDefaults()
+    {
+        $stateDefaults = parent::getSchemaStateDefaults();
+        $config = $this->getEditorConfig();
+        $stateDefaults['data']['config'] = $config->getAttributes();
+        $stateDefaults['data']['editorjs'] = null;
+        if (method_exists($config, 'getScriptURL')) {
+            $stateDefaults['data']['editorjs'] = $config->getScriptURL();
+        }
```

Comment:
Perhaps you could overload this method in `TinyMCEConfig` and add in the part that's specific to it there?

Reply:
ooh, that's a much better idea 👍

Reply:
oh, whoops... this is `HTMLEditorField`, not `HTMLEditorConfig` ;) There is no `TinyMCEField` to override the method in.

Reply:
I see. I think we can probably to this in a more structured way still though. Perhaps `HTMLEditorConfig` and `TinyMCEConfig` need a new method like `getSchemaData()` added?

Reply:
Yeah I guess the options are:

1) Config provides its own schema data, irrespective of the format of the HTMLEditorField's schema -> call it `getSchemaData` and return an isolated array
2) Config augments the existing HTMLEditorField's schema data (minor coupling) -> call it `augmentSchemaData` and pass the existing array by reference for modification

Reply:
Thinking about the way it's currently structured...
1) would require another level of data e.g. `this.props.data.config.attributes['data-whatever']` to keep things nicely separated.
2) sounds like a reasonably bad idea, unless it was a mixture of getting an independent array, then recursively merging the two results in some fashion (i.e. perhaps keeping the Field's data as immutable in a similar way to `getSchemaData` on `FormField`)

Reply:
`data` key on schema is freeform anyway, so I just left everything operating as it was (a true refactor).